facebook/facebook-for-woocommerce
# Add rich text description to woo product sync to meta

### Add Rich Text Description to Woo Product Sync with Meta

#### Description
This PR introduces a new feature to the Facebook WooCommerce Plugin, allowing users to add rich text descriptions to their products. With this update, we can seamlessly synchronize the rich text description field with the Facebook Commerce Manager platform.

It also adds functionality for a WYSIWYG input box for the Facebook Description field in the plugin 

#### Changes
- Updated the product sync logic to handle rich text formatting.
- Modified the API payload to include the rich text description field.
- Tested for compatibility with common formatting options (bold, italic, lists, links, etc.).

#### Benefits
- Improves the visual appeal of product descriptions on Meta.
- Ensures consistent branding and formatting across platforms.
- Enhances user engagement by presenting more structured and detailed product information.

#### Testing Instructions
1. Enable the WooCommerce to Meta product sync.
2. Create or edit a product in WooCommerce with a rich text description (e.g., headings, bold, italic, lists).
3. Sync the product and verify that the description appears correctly formatted on Meta.


#### Screenshots
<img width="1244" alt="Screenshot 2024-11-04 at 11 21 23" src="https://github.com/user-attachments/assets/eb1c1fbb-5734-4c27-b710-1e031f1a58e6">
<img width="601" alt="Screenshot 2024-11-04 at 11 22 26" src="https://github.com/user-attachments/assets/a40f84e8-3127-4b62-a87d-c172ab257a98">
<img width="753" alt="Screenshot 2024-11-04 at 11 22 41" src="https://github.com/user-attachments/assets/75063389-5a78-4ae2-a73f-c2a392427472">
<img width="740" alt="Screenshot 2024-12-24 at 18 24 52" src="https://github.com/user-attachments/assets/e011e53e-9892-4e06-ace1-bfd7e22dab22" />

---

#### Additional Notes
Please let me know if there are any edge cases or scenarios that should be tested further.
